tests/store/test_dbfs_artifact_repo.py

```diff
@@ -0,0 +1,127 @@
+
+
+class TestDbfsArtifactRepository(object):
+    def test_init_validation_and_cleaning(self):
+        repo = DbfsArtifactRepository('dbfs:/test/', {})
+        assert repo.artifact_uri == 'dbfs:/test'
+        with pytest.raises(AssertionError):
+            DbfsArtifactRepository('s3://test', {})
+
+    @pytest.mark.parametrize("artifact_path,expected_endpoint", [
+        (None, '/dbfs/test/test.txt'),
+        ('output.txt', '/dbfs/test/output.txt'),
+        ('subdir/output.txt', '/dbfs/test/subdir/output.txt'),
+    ])
+    def test_log_artifact(self, dbfs_artifact_repo, test_file, artifact_path, expected_endpoint):
+        with mock.patch('mlflow.store.dbfs_artifact_repo.http_request') as http_request_mock:
+            dbfs_artifact_repo.log_artifact(test_file.strpath, artifact_path)
+            assert http_request_mock.called
+            _, kwargs = http_request_mock.call_args
+            assert kwargs['endpoint'] == expected_endpoint
+
+    def test_log_artifact_empty(self, dbfs_artifact_repo, test_file):
+        with pytest.raises(IllegalArtifactPathError):
+            dbfs_artifact_repo.log_artifact(test_file.strpath, '')
+
+    @pytest.mark.parametrize("artifact_path", [
+        None,
+        '', # should behave like '/' and exclude base name of logged_dir
+        # We should add '.',
+    ])
+    def test_log_artifacts(self, dbfs_artifact_repo, test_dir, artifact_path):
+        with mock.patch('mlflow.store.dbfs_artifact_repo.http_request') as http_request_mock:
+            dbfs_artifact_repo.log_artifacts(test_dir.strpath, artifact_path)
+            basename = test_dir.basename
+            assert_endpoints(http_request_mock.call_args_list, {
+                '/dbfs/test/%s/subdir/test.txt' % basename,
+                '/dbfs/test/%s/test.txt' % basename
+            })
+
+    @pytest.mark.parametrize("artifact_path,expected_endpoints", [
+        ('a', {'/dbfs/test/a/subdir/test.txt', '/dbfs/test/a/test.txt'}),
+        ('a/', {'/dbfs/test/a/subdir/test.txt', '/dbfs/test/a/test.txt'}),
+        ('/', {'/dbfs/test/subdir/test.txt', '/dbfs/test/test.txt'}),
+    ])
+    def test_log_artifacts_with_artifact_path(self, dbfs_artifact_repo, test_dir, artifact_path,
+                                              expected_endpoints):
+        with mock.patch('mlflow.store.dbfs_artifact_repo.http_request') as http_request_mock:
+            dbfs_artifact_repo.log_artifacts(test_dir.strpath, artifact_path)
+            assert_endpoints(http_request_mock.call_args_list, expected_endpoints)
+
+    def test_list_artifacts(self, dbfs_artifact_repo):
+        with mock.patch('mlflow.store.dbfs_artifact_repo.http_request') as http_request_mock:
+            http_request_mock.return_value.text = json.dumps(LIST_ARTIFACTS_RESPONSE)
+            artifacts = dbfs_artifact_repo.list_artifacts()
+            assert len(artifacts) == 2
+            assert artifacts[0].path == '/a.txt'
```

Comment:
Q: can the UI handle these leading slashes? e.g. I think the LocalArtifactRepository will return paths without leading slashes from [list_artifacts](https://github.com/mlflow/mlflow/blob/master/mlflow/store/local_artifact_repo.py#L30)

Reply:
Good point, probably not.